Add support for "include_segment_file_sizes" in indices stats REST handler

spiegela opened the pull request:
Fix for #21878

Adds `include_segment_file_sizes` to acceptable parameters in the `RestIndicesStatsAction` REST handler.

Thanks!

jasontedor commented on core/src/main/java/org/elasticsearch/rest/action/admin/indices/RestIndicesStatsAction.java:
```diff
@@ -157,7 +158,7 @@ public boolean canTripCircuitBreaker() {
     }
 
-    private static final Set<String> RESPONSE_PARAMS = Collections.singleton("level");
+    private static final Set<String> RESPONSE_PARAMS = new HashSet<String>(Arrays.asList("level", "include_segment_file_sizes"));
```
This isn't the right fix, it's not a response parameter (it controls the request that the node client sends). Rather, this parameter needs to be consumed (and parsed as a boolean).

spiegela replied:
Gotcha. I see some examples of that now using the `request.paramAsBoolean`.  I'll correct it this way, and correct it in the node stats, too.

jasontedor commented:
I left a comment requesting a different approach. The same fix needs to be applied to node stats too.

jasontedor commented:
@spiegela Also, I think I'd prefer the tests to be REST tests as opposed to unit tests; please let me know if you can make these changes or if I can help guide you through this.

spiegela commented:
Thanks, @jasontedor.  I can make the fix by consuming the parameter pretty quickly. Can you point me to a REST test example?

jasontedor commented:
Look at `rest-api-spec/src/main/resources/rest-api-spec/test/indices.stats` and similarly for nodes stats. And you can poke around in the sibling folders for more examples. Feel free to ping upon any questions.

spiegela commented:
@jasontedor I've fixed the issue by consuming the param as a bool, and switched the testing to the REST tests. I've also corrected the same issue in the node stats.

jasontedor commented on rest-api-spec/src/main/resources/rest-api-spec/api/indices.stats.json:
```diff
@@ -52,6 +52,11 @@
+        },
+        "include_segment_file_sizes": {
+          "type": "boolean",
+          "description": "Whether report the aggregated disk usage of each one of the Lucene index files.",
```
Can you fix the grammar here, remove the period, and add a remark that it only applies if segments stats are requested?

spiegela commented:
@jasontedor Should have waited till the morning to push and caught both of those. I also cleaned up some unused imports.

jasontedor commented:
@spiegela I think you force pushed? The review process is easier if we go commit by commit and we can squash at the end on our side when we integrate. No worries this time, just a heads up for the future. :smile:

jasontedor commented:
ok to test

jasontedor commented:
I will merge this soon.

jasontedor commented:
Thanks @spiegela!

spiegela commented:
Sure thing!
